src/utils/api/account/lsk.js

```diff
@@ -79,15 +80,21 @@ export const getAccount = async ({
       params: normParams,
     });
 
-    return response.data[0];
+    if (response.data[0]) {
+      const account = { ...response.data[0] };
+      const isAccountUninitialized = !account.summary?.publicKey;
+      if (isAccountUninitialized) {
+        const publicKey = params.publicKey ?? extractPublicKey(params.passphrase);
+        account.summary.publicKey = publicKey;
+      }
+
+      return account;
+    }
   } catch (e) {
     // eslint-disable-next-line no-console
     console.log('Lisk account not found.');
-    let publicKey = params.publicKey;
-    if (!publicKey && params.passphrase) {
-      publicKey = extractPublicKey(params.passphrase);
-    }
 
+    const publicKey = params.publicKey ?? extractPublicKey(params.passphrase);
```

Comment:
If the method is called with address and it returned 404, this assignment will throw. The previous one was better, although it didn't look as dope.

Reply:
Can you please explain 1. why do you think the assignment will throw 2. why the previous one was better (except readability)?